pkg/runctl/controller.go

```diff
@@ -342,35 +342,35 @@ func (c *Controller) syncHandler(key string) error {
 
 	ctx = klog.NewContext(ctx, logger)
 
-	doReturn, err := c.handlePipelineRunFinalizerAndDeletion(ctx, pipelineRun)
+	// the configuration should be loaded once per sync to avoid inconsistencies
+	// in case of concurrent configuration changes
+	pipelineRunsConfig, doReturn, err := c.ensurePipelineRunsConfig(ctx, pipelineRun)
 	if doReturn || err != nil {
 		return err
 	}
```

Comment:
This was _intentionally not_ done before `c.handlePipelineRunFinalizerAndDeletion()` because it would drastically slow down processing of potentially _many_ pipeline runs that don't need processing anymore, as it happens in initial syncs and resyncs.

-> Revert

The right approach is to introduce an config loader object created for each execution of `syncHandler()`. It has a method to return the config, and caches it after it has been loaded the first time. The load method gets called where the config is required.

Reply:
Changed coding to have lazy loading of pipeline run configuration.